File: mybatis-generator-core/src/main/java/com/wvkity/mybatis/code/make/utils/PropertiesUtil.java
package com.wvkity.mybatis.code.make.utils;

import java.io.File;
import java.io.FileInputStream;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.nio.charset.StandardCharsets;
import java.util.HashMap;
import java.util.Map;
import java.util.Optional;
import java.util.Properties;

/**
 * properties文件工具
 * @author wvkity
 */
public final class PropertiesUtil {

    private PropertiesUtil() {
    }

    /**
     * 加载properties文件
     * @param file properties文件
     * @return 配置
     */
    public static Map<String, String> load(final File file) {
        try (final InputStream fs = new FileInputStream(file)) {
            return load(fs);
        } catch (Exception ignore) {
            // ignore
        }
        return new HashMap<>(0);
    }

    /**
     * 加载properties文件
     * @param inputStream properties文件流
     * @return 配置
     */
    public static Map<String, String> load(final InputStream inputStream) {
        return Optional.ofNullable(inputStream).map(it -> {
            final Map<String, String> map = new HashMap<>();
            try (final InputStreamReader isr = new InputStreamReader(inputStream, StandardCharsets.UTF_8)) {
                final Properties props = new Properties();
                props.load(isr);
                props.stringPropertyNames().forEach(k ->
                        map.put(k, props.getProperty(k, "")));
            } catch (Exception ignore) {
                // ignore
            }
            return map;
        }).orElse(new HashMap<>(0));
    }
}

File: mybatis-generator-gui/src/main/kotlin/com/wkit/lost/mybatis/generator/utils/PropertiesUtil.kt
package com.wkit.lost.mybatis.generator.utils

import org.apache.logging.log4j.LogManager
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*
import kotlin.collections.HashMap

class PropertiesUtil {

    companion object {

        private val LOG = LogManager.getLogger(PropertiesUtil)

        fun loadProperties(inputStream: InputStream?): MutableMap<String, String> {
            return inputStream?.run {
                return inputStream.use {
                    val map = HashMap<String, String>()
                    try {
                        val properties = Properties()
                        properties.load(InputStreamReader(inputStream, "utf-8"))
                        val keys = properties.stringPropertyNames()
                        keys.forEach {
                            map[it] = properties.getProperty(it, "")
                        }
                    } catch (e: Exception) {
                        LOG.error("Failed to read configuration file: {}", e.message, e)
                    }
                    map
                }
            } ?: run {
                HashMap<String, String>()
            }
        }
    }
}